addons/link_tracker/tests/test_merge_utm_campaign.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import SavepointCase
+
+
+class TestMergeUtmCampaign(SavepointCase):
+    def setUp(self):
+        super(TestMergeUtmCampaign, self).setUp()
+        self.utm_source = self.env['utm.source'].create({
+            'name': 'utm source'
+        })
+        self.utm_medium = self.env['utm.medium'].create({
+            'name': 'utm medium'
+        })
+        self.utm_campaign_1 = self.env['utm.campaign'].create({
+            'name': 'utm campaign 1'
+        })
+        self.utm_campaign_2 = self.env['utm.campaign'].create({
+            'name': 'utm campaign 2'
+        })
+        self.utm_campaign_3 = self.env['utm.campaign'].create({
+            'name': 'utm campaign 3'
+        })
+        self.link_tracker_1 = self.env['link.tracker'].create({
+            'url': 'www.dodd.com',
+            'campaign_id': self.utm_campaign_1.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_2 = self.env['link.tracker'].create({
+            'url': 'www.dodd.com',
+            'campaign_id': self.utm_campaign_2.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_3 = self.env['link.tracker'].create({
+            'url': 'www.odoo.com',
+            'campaign_id': self.utm_campaign_3.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_click_1 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_1.id
+        })
+        self.link_tracker_click_2 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_2.id
+        })
+        self.link_tracker_click_3 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_3.id
+        })
+
+    def test_merge(self):
+        """ Merge 2 campaigns and check that everything is correctly updated.
+        Unneeded campaigns should be deactivated, and contain reference to the remaining one.
+        (see 'utm.py'#_merge_utm_campaigns) """
+        (self.utm_campaign_1 + self.utm_campaign_2)._merge_utm_campaigns(self.utm_campaign_1)
+
+        self.assertEqual(self.utm_campaign_2.active, False, 
+            "After merging utm.campaigns, unneeded campaigns should be archived. ")
+        self.assertEqual(self.utm_campaign_2.reference_utm_campaign_id, self.utm_campaign_1, 
+            "After merging utm.campaigns, deactivated utm.campaign should refer to the merged campaign. ")
```

Comment:
This test does the same operation as the next one (merge campaign_1 and campaign_2 together).

You can move your assertions to the next test and remove this one.